### DISCUSSION TITLE
What is a proper way to run backfill with KubernetesExecutor

### DISCUSSION THREAD

**Discussion by ashlee3209** [apache/airflow#23216] [Q&A]
We are using the airflow 2.2.4 with KubernetesExecutor and we have a requirement to run tasks for some dags with execution date before the start date of the dags which could be done with running airflow backfill using cli. 

I first tried to run this command in the airflow scheduler pod and the backfill tasks will be created as new pods in the cluster as the normal scheduled ones. 

Then I tried to create a backfill airflow dag which runs this comand in a subprocesshook and then it happens that the backfill tasks will not create new pods in the cluster but running inside the backfill dag pod. Is that an expected behavior? How could I make those tasks to be run as new pods? Because every pod has a limitation of cpu and memory, a single pod cannot run multiple tasks at the same time which could be killed due to out of memory.

I am not sure if I have explained myself clearly. Does anybody have the same issue and how do you solve that?

Thank you!

**Comment by potiuk** [comment]
> I first tried to run this command in the airflow scheduler pod and the backfill tasks will be created as new pods in the cluster as the normal scheduled ones.

What's the problem with that - it looks that this should work.

**Reply by ashlee3209** [reply]
No, it is not working. The backfill tasks got stuck even though the pod finished in k8s. See my comment in another discussion: https://github.com/apache/airflow/discussions/21047#discussioncomment-2608664
